Sources/PrimerSDK/Classes/Core/3DS/3DSService.swift

```diff
@@ -93,3 +93,62 @@ class ThreeDSService: ThreeDSServiceProtocol {
     }
     
 }
+
+class MockThreeDSService: ThreeDSServiceProtocol {
+    
+    var response: Data?
+    let throwsError: Bool = false
+    var isCalled: Bool = false
+    
+    init(with response: Data? = nil) {
+        self.response = response
+    }
+    
+    func initializeSDK(_ sdk: ThreeDSSDKProtocol, completion: @escaping (Result<Void, Error>) -> Void) {
+        sdk.initializeSDK(completion: completion)
+    }
+    
+    func sdkAuth(sdk: ThreeDSSDKProtocol, cardNetwork: CardNetwork, protocolVersion: ThreeDS.ProtocolVersion, completion: @escaping (Result<Transaction, Error>) -> Void) {
+        sdk.sdkAuth(cardNetwork: cardNetwork, protocolVersion: protocolVersion, completion: completion)
+    }
+    
+    func beginRemoteAuth(paymentMethodToken: PaymentMethodToken, threeDSecureBeginAuthRequest: ThreeDS.BeginAuthRequest, completion: @escaping (Result<ThreeDS.BeginAuthResponse, Error>) -> Void) {
+        let state: AppStateProtocol = DependencyContainer.resolve()
+        
+        guard let clientToken = state.decodedClientToken else {
+            return completion(.failure(PrimerError.vaultFetchFailed))
+        }
+        
+        let api = MockPrimerAPIClient()
+        DependencyContainer.register(api as PrimerAPIClientProtocol)
```

Comment:
[Suggestion]
Wouldn't it be better to do a dependency injection, and having this on the test rather than in `MockThreeDSService`?